pilot/pkg/xds/ads.go

```diff
@@ -219,9 +219,11 @@ func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *C
 			&model.WatchedResource{TypeUrl: req.TypeUrl, ResourceNames: req.ResourceNames},
 			&model.PushRequest{Full: true, Push: con.proxy.LastPushContext})
 	}
-	if s.StatusReporter != nil {
-		s.StatusReporter.RegisterEvent(con.conID, req.TypeUrl, req.ResponseNonce)
+
+	if s.StatusReporter != nil && AllTrackingEventTypes.Contains(req.TypeUrl) {
```

Comment:
> > The most important factor, by removing it, we can eliminate allocating space for watched typeurls during each push.

Is this the motivation? To remove allocating a single array of ~8 strings? If so, this seems like a  premature optimization?

Reply:
```
Specifically, imagine this:

Push LDS only at v1
Push CDS only at v2
ACK CDS at v2
ACK LDS at v1
```

For ads it is using a single stream,  seems not possible.


This is not just the only motivation. As the debug interface shows ` "/debug/config_distribution", "Version status of all Envoys connected to this Pilot instance",`
While previously it is not right, it records those versions it sends, regardless of envoy received or loaded successfully.

So we can treat this as a bug fix too

Reply:
I do not believe the types are coupled in envoy/xds in such a way that the above flow is not possible

Reply:
Do you have some good solution on this, otherwise let me think that

Reply:
Yes, I would just not make this change at all. I do not understand the motivation for this change at all , what problem are we trying to solve?

Reply:
I think we can just clarify in the comments that if the version is not relevant to the type it will be considered sent+acked

although I think that also has an issue even - if I do

* push type1 (never ack)
* push type2 

we will report all on the 2nd push and never notice the missed ack.

The overall design has many flaws, I would prefer to keep the current state (flawed) than to transition to a new state that is also flawed, unless someone is willing to holistically own the feature end to end and drive it out of alpha

Reply:
OK, sounds goog to me